src/System.Runtime/tests/System/GuidTests.cs

```diff
@@ -372,11 +419,59 @@ public static void ToString(Guid guid, string format, string expected)
             Assert.Equal(expected, formattable.ToString(format, null));
         }
 
-        [Fact]
-        public static void ToString_InvalidFormat_ThrowsFormatException()
+        [Theory]
+        [InlineData("Y")]
+        [InlineData("XX")]
+        public static void ToString_InvalidFormat_ThrowsFormatException(string format)
+        {
+            Assert.Throws<FormatException>(() => s_testGuid.ToString(format));
+        }
+
+        [Theory]
+        [InlineData("Y")]
+        [InlineData("XX")]
+        public static void TryFormat_InvalidFormat_ThrowsFormatException(string format)
+        {
+            Assert.Throws<FormatException>(() => s_testGuid.TryFormat(new Span<char>(), out int charsWritten, format));
+        }
+
+        [Theory]
+        [MemberData(nameof(ToString_TestData))]
+        public static void TryFormat_ReturnsTrue(Guid guid, string format, string expected, int length)
+        {
+            Assert.True(guid.TryFormat(new Span<char>(new char[length]), out int charsWritten, format));
+        }
+
+        [Theory]
+        [MemberData(nameof(ToString_TestData))]
+        public static void TryFormat_ReturnsFalse(Guid guid, string format, string expected, int length)
+        {
+            Assert.False(guid.TryFormat(new Span<char>(new char[length - 1]), out int charsWritten, format));
+        }
+
+        [Theory]
+        [MemberData(nameof(ToString_TestData))]
+        public static void TryFormat_CharsWritten(Guid guid, string format, string expected, int length)
+        {
+            guid.TryFormat(new Span<char>(new char[length]), out int charsWritten, format);
+            Assert.Equal(length, charsWritten);
+        }
+
+        [Theory]
+        [MemberData(nameof(ToString_TestData))]
+        public static void TryFormat_CharsWritten_EqualsZero(Guid guid, string format, string expected, int length)
+        {
+            guid.TryFormat(new Span<char>(new char[length - 1]), out int charsWritten, format);
+            Assert.Equal(0, charsWritten);
+        }
+
+        [Theory]
+        [MemberData(nameof(ToString_TestData))]
+        public static void TryFormat_Valid(Guid guid, string format, string expected, int length)
         {
-            Assert.Throws<FormatException>(() => s_testGuid.ToString("Y")); // Invalid format
-            Assert.Throws<FormatException>(() => s_testGuid.ToString("XX")); // Invalid format
+            char[] chars = new char[length];
+            guid.TryFormat(new Span<char>(chars), out int charsWritten, format);
```

Comment:
For all of these Try calls being added, we should assert the result is as expected.